src/screens/Services/Services.js

```diff
@@ -227,39 +230,69 @@ class ServicesScreen extends React.Component<Props, State> {
         title: 'Buy with Wyre (Non-EU)',
         body: 'Buy Now',
         action: () => {
-          this.handleBuyCryptoAction(address => `${SENDWYRE_WIDGET_URL}?${querystring.stringify({
-            accountId: SENDWYRE_ACCOUNT_ID,
-            dest: `ethereum:${address}`,
-            redirectUrl: SENDWYRE_RETURN_URL,
-          })}`);
+          const address = this.getCryptoPurchaseAddress();
+          if (address === null) return;
+          this.tryOpenCryptoPurchaseUrl(wyreWidgetUrl(address));
+        },
+      });
+    }
+
+    if (isAltalixEnabled && isAltalixAvailable) {
+      buyCryptoServices.push({
+        key: 'altalix',
+        title: 'Buy with Altalix',
+        body: 'Buy Now',
+        action: async () => {
+          const { user: { walletId }, getApi } = this.props;
+          const address = this.getCryptoPurchaseAddress();
+          if (address === null) return;
+          this.tryOpenCryptoPurchaseUrl(await altalixWidgetUrl({
+            walletId,
+            address,
+            sellCurrency: 'EUR',
+            buyCurrency: 'ETH',
+
+            // The amount is adjustable in the Altalix app, but the link won't work
+            // if the initial value is 0
+            buyAmount: 0.02,
+          }, getApi()));
         },
       });
     }
 
     return buyCryptoServices;
   }
 
-  handleBuyCryptoAction = (getUrlWithAddress: string => string) => {
+  getCryptoPurchaseAddress = (): string | null => {
     const { accounts, smartWalletState } = this.props;
 
     const activeAccount = getActiveAccount(accounts);
     const smartWalletStatus = getSmartWalletStatus(accounts, smartWalletState);
 
     if (!smartWalletStatus.hasAccount) {
       this.setState({ buyCryptoModalMessage: ACCOUNT_MSG.NO_SW_ACCOUNT });
-      return;
+      return null;
     }
 
     if (!activeAccount || !checkIfSmartWalletAccount(activeAccount)) {
       this.setState({ buyCryptoModalMessage: ACCOUNT_MSG.SW_ACCOUNT_NOT_ACTIVE });
-      return;
+      return null;
     }
 
-    const address: string = getAccountAddress(activeAccount);
-    const url = getUrlWithAddress(address);
-    openInAppBrowser(url);
+    return getAccountAddress(activeAccount);
   }
 
+  tryOpenCryptoPurchaseUrl = (url: string | null) => {
+    if (url) {
+      openInAppBrowser(url);
```

Comment:
Does `openInAppBrowser` return an error or something that we can report when `openInAppBrowser` doesn't work?

Reply:
`openInAppBrowser` takes care of reporting to Sentry, but I imagine we'd like to show the same error message as with failed URL construction -- I'll make it return a rejected promise on top of that.